### PR TITLE
[FIX] web: prevent upload of large files

### PR DESCRIPTION
**Pull request by brboi**
(NB: this is a backport of commit [2a72028](https://github.com/odoo/odoo/commit/2a72028ce64d129ea7cd7c8ce84d1e54f966bbdf))

This commit ensures that files uploaded using the FileInput, upload service or AttachDocumentWidget will not exceed the maximum file size of the context in size before sending them to the server. It also moves some utils from the formatters file to the numbers file so that one can use them without requiring the view assets.

### CODE REVIEW

**Comment by brboi** [comment]
@fw-bot up to saas-16.4

File: addons/web_editor/static/src/components/upload_progress_toast/upload_service.js [hunk 1]
```diff
@@ -3,14 +3,16 @@
...
+import { humanNumber } from "@web/core/utils/numbers";
 
 import { reactive } from "@odoo/owl";
 
 export const AUTOCLOSE_DELAY = 3000;
 
 export const uploadService = {
-    dependencies: ['rpc'],
-    start(env, { rpc }) {
+    dependencies: ['rpc', 'notification'],
```
Comment (aab-odoo) [inline addons/web_editor/static/src/components/upload_progress_toast/upload_service.js:14]: this is a breaking change

**Review by aab-odoo — COMMENTED** [review]
I would try to reduce the diff to the strict minimum for 16.0 as it's a bit scary like this. Also there're technically breaking changes (even though not likely to happen) with the new dependency to the notification service.

File: addons/web/static/src/core/file_input/file_input.js [hunk 2]
```diff
@@ -25,6 +26,7 @@ import { Component, onMounted, useRef } from "@odoo/owl";
 export class FileInput extends Component {
     setup() {
         this.http = useService("http");
+        this.notification = useService("notification");
```
Comment (aab-odoo) [inline addons/web/static/src/core/file_input/file_input.js:29]: this is breaking as well

**Review by aab-odoo — APPROVED** [review]

**Review by aab-odoo — APPROVED** [review]
robodoo r+

**Comment by bobby-sapentia** [comment]
This breaks the project sharing chatter:

```UncaughtPromiseError > TypeError
Uncaught Promise > Cannot read properties of undefined (reading 'length')
TypeError: Cannot read properties of undefined (reading 'length')
    at PortalFileInput.uploadFiles (https://sapentia-corporate.odoo.com/web/assets/debug/project.webclient.js:7655:26) (/web/static/src/core/file_input/file_input.js:54)
    at https://sapentia-corporate.odoo.com/web/assets/debug/project.webclient.js:118278:27 (/project/static/src/project_sharing/components/portal_file_input/portal_file_input.js:28)
    at Array.map (<anonymous>)
    at PortalFileInput.uploadFiles (https://sapentia-corporate.odoo.com/web/assets/debug/project.webclient.js:118276:19) (/project/static/src/project_sharing/components/portal_file_input/portal_file_input.js:26)
    at PortalFileInput.onFileInputChange (https://sapentia-corporate.odoo.com/web/assets/debug/project.webclient.js:7688:43) (/web/static/src/core/file_input/file_input.js:87)
    at Object.mainEventHandler (https://sapentia-corporate.odoo.com/web/assets/debug/web.assets_common.js:22237:25) (/web/static/lib/owl/owl.js:5843)
    at HTMLInputElement.listener (https://sapentia-corporate.odoo.com/web/assets/debug/web.assets_common.js:16751:20) (/web/static/lib/owl/owl.js:357)
```
image

Reverting to an older version of Odoo before the commit date on Odoo.sh fixes it, but is not a long-term solution.
Could you please amend this?

**Pull request closed without merging**